library/extractor/src/main/java/com/google/android/exoplayer2/extractor/ogg/OpusReader.java

```diff
@@ -69,13 +76,41 @@ protected boolean readHeaders(ParsableByteArray packet, long position, SetupData
               .setSampleRate(OpusUtil.SAMPLE_RATE)
               .setInitializationData(initializationData)
               .build();
+
       headerRead = true;
       return true;
     } else {
       checkNotNull(setupData.format); // Has been set when the header was read.
-      boolean headerPacket = packet.readInt() == OPUS_CODE;
-      packet.setPosition(0);
-      return headerPacket;
+
+      // Handle the comments frame if we are working with one.
+      boolean isComment;
+
+      if (packet.bytesLeft() < OPUS_TAG_SIGNATURE.length) {
+        isComment = false;
+      } else {
+        byte[] header = new byte[OPUS_TAG_SIGNATURE.length];
+        packet.readBytes(header, 0, OPUS_TAG_SIGNATURE.length);
+        isComment = Arrays.equals(header, OPUS_TAG_SIGNATURE);
+      }
+
+      if (isComment) {
+        VorbisUtil.CommentHeader commentHeader = VorbisUtil.readVorbisCommentHeader(
+            packet, false, false);
+
+        @Nullable Metadata vorbisMetadata = VorbisUtil.buildMetadata(
+            Arrays.asList(commentHeader.comments)
+        );
+
+        setupData.format = setupData.format
+            .buildUpon()
+            .setMetadata(vorbisMetadata)
```

Comment:
I think it would be best to append. I'm considering adding my own extension [not to be merged] to my app's variant that would add the "base gain" header field to the Metadata instance, and I would rather not want a gotcha where the `Metadata` is overwritten like that.

Reply:
Resolved with b919161.